ui/src/store/resource.test.ts

```diff
@@ -262,4 +264,201 @@ describe('Store functions', () => {
+      () => !store.isLoading,
+      () => {
+        expect(store.isLoading).toBe(false);
+        expect(store.resources.size).toBe(7);
+
+        const webURL = store.resources.get('aws-cli')?.webURL;
+
+        expect(webURL).toBe('https://github.com/tektoncd/catalog/tree/master/task/aws-cli/0.1/');
+        done();
+      }
+    );
+  });
+  it('it should return short description', (done) => {
+    const store = ResourceStore.create(
+      {},
+      {
+        api,
+        categories: CategoryStore.create({}, { api })
+      }
+    );
+    expect(store.isLoading).toBe(true);
+    when(
+      () => !store.isLoading,
+      () => {
+        expect(store.isLoading).toBe(false);
+        expect(store.resources.size).toBe(7);
+
+        const description = store.resources.get('aws-cli')?.shortDescription;
+
+        expect(description).toBe(
+          'This task performs operations on Amazon Web Services resources using aws.'
+        );
+        done();
+      }
+    );
+  });
+  it('it should return detail description', (done) => {
+    const store = ResourceStore.create(
+      {},
+      {
+        api,
+        categories: CategoryStore.create({}, { api })
+      }
+    );
+    expect(store.isLoading).toBe(true);
+    when(
+      () => !store.isLoading,
+      () => {
+        expect(store.isLoading).toBe(false);
+        expect(store.resources.size).toBe(7);
+
+        const description = store.resources.get('buildah')?.detailDescription;
+
+        expect(description).toBe(
+          "Buildah Task builds source into a container image using Project Atomic's Buildah build tool.It uses Buildah's support for building from Dockerfiles, using its buildah bud command.This command executes the directives in the Dockerfile to assemble a container image, then pushes that image to a container registry."
+        );
+        done();
+      }
+    );
+  });
+  it('it should return install command', (done) => {
+    const store = ResourceStore.create(
+      {},
+      {
+        api,
+        categories: CategoryStore.create({}, { api })
+      }
+    );
+    expect(store.isLoading).toBe(true);
+    when(
+      () => !store.isLoading,
+      () => {
+        expect(store.isLoading).toBe(false);
+        expect(store.resources.size).toBe(7);
+
+        const installCmd = store.resources.get('aws-cli')?.installCommand;
+
+        expect(installCmd).toBe(
+          'kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/aws-cli/0.1/aws-cli.yaml'
+        );
+        done();
+      }
+    );
+  });
+  it('update versions list for buildah resource', (done) => {
+    const store = ResourceStore.create(
+      {},
+      {
+        api,
+        categories: CategoryStore.create({}, { api })
+      }
+    );
+
+    expect(store.isLoading).toBe(true);
+
+    when(
+      () => !store.isLoading,
+      () => {
+        expect(store.resources.size).toBe(7);
+        expect(getSnapshot(store.resources)).toMatchSnapshot();
```

Comment:
Why are we taking snapshot here again ??? Are versions added ??

Reply:
sorry I placed this at wrong place. Will fix this. Thanks for pointing this out :smile: